### ISSUE TITLE
Error installing teleport-ent-updater from yum repo

### ISSUE THREAD

**Issue by webvictim** [gravitational/teleport#27729]
```
gus@hades:~$ sudo yum install teleport-ent-updater
Last metadata expiration check: 0:00:11 ago on Mon 12 Jun 2023 10:46:49 ADT.
Dependencies resolved.
======================================================================================================================================================
 Package                                      Architecture                   Version                           Repository                        Size
======================================================================================================================================================
Installing:
 teleport-ent-updater                         x86_64                         13.1.0-1                          teleport                          12 k

Transaction Summary
======================================================================================================================================================
Install  1 Package

Total download size: 12 k
Installed size: 15 k
Is this ok [y/N]: y
Downloading Packages:
teleport-ent-updater-13.1.0.amd64.rpm                                                                                  19 kB/s |  12 kB     00:00
------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                  19 kB/s |  12 kB     00:00
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                              1/1
  Installing       : teleport-ent-updater-13.1.0-1.x86_64                                                                                         1/1
  Running scriptlet: teleport-ent-updater-13.1.0-1.x86_64                                                                                         1/1
warning: %post(teleport-ent-updater-13.1.0-1.x86_64) scriptlet failed, signal 1

Error in POSTIN scriptlet in rpm package teleport-ent-updater
```

Teleport version: 13.1.0

Extra debugging:
```
gus@hades:~$ cat /etc/os-release
NAME="Fedora Linux"
VERSION="37 (Server Edition)"
ID=fedora
VERSION_ID=37
VERSION_CODENAME=""
PLATFORM_ID="platform:f37"
PRETTY_NAME="Fedora Linux 37 (Server Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:37"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f37/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=37
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=37
SUPPORT_END=2023-11-14
VARIANT="Server Edition"
VARIANT_ID=server
```

**Comment by webvictim** [comment]
For extra context, this failure to start the timer also terminated my SSH session, because it restarted the Teleport process that I was using to connect to the system. I presume that the post-install command is malformed enough that it targetted `teleport.service` instead of `teleport-upgrade.timer`, which seems dangerous.

**Comment by webvictim** [comment]
FYI this also affects the `teleport-ent-updater` apt package, and has the same symptom of restarting the `teleport` daemon and not enabling `teleport-upgrade.timer`.

```
root@cassandra:/etc/teleport-upgrade.d# sudo apt-get -y install teleport-ent-updater
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  teleport-ent-updater
0 upgraded, 1 newly installed, 0 to remove and 11 not upgraded.
Need to get 7,006 B of archives.
After this operation, 15.4 kB of additional disk space will be used.
Get:1 https://apt.releases.teleport.dev/ubuntu focal/stable/v13 arm64 teleport-ent-updater arm64 13.1.5 [7,006 B]
Fetched 7,006 B in 0s (18.4 kB/s)
Selecting previously unselected package teleport-ent-updater.
(Reading database ... 49688 files and directories currently installed.)
Preparing to unpack .../teleport-ent-updater_13.1.5_arm64.deb ...
Unpacking teleport-ent-updater (13.1.5) ...
Setting up teleport-ent-updater (13.1.5) ...

client_loop: send disconnect: Broken pipe
```

**Comment by bernardjkim** [comment]
It doesn't seem to be a problem anymore as of teleport-ent-updater 13.2.1

Upgraded successfully from 13.1.4 -> 13.2.1 without disconnecting from tsh ssh and timer unit was enabled.
```
$ sudo yum install teleport-ent-updater-13.2.1-1
Last metadata expiration check: 0:03:08 ago on Thu 20 Jul 2023 04:35:32 PM UTC.
Dependencies resolved.
===================================================================================================================================================================================
 Package                                             Architecture                          Version                                   Repository                               Size
===================================================================================================================================================================================
Upgrading:
 teleport-ent-updater                                x86_64                                13.2.1-1                                  teleport                                 13 k

Transaction Summary
===================================================================================================================================================================================
Upgrade  1 Package

Total download size: 13 k
Is this ok [y/N]: y
Downloading Packages:
teleport-ent-updater-13.2.1.amd64.rpm                                                                                                               47 kB/s |  13 kB     00:00
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                               46 kB/s |  13 kB     00:00
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                           1/1
  Upgrading        : teleport-ent-updater-13.2.1-1.x86_64                                                                                                                      1/2
  Running scriptlet: teleport-ent-updater-13.2.1-1.x86_64                                                                                                                      1/2
  Cleanup          : teleport-ent-updater-13.1.4-1.x86_64                                                                                                                      2/2
  Running scriptlet: teleport-ent-updater-13.1.4-1.x86_64                                                                                                                      2/2
  Verifying        : teleport-ent-updater-13.2.1-1.x86_64                                                                                                                      1/2
  Verifying        : teleport-ent-updater-13.1.4-1.x86_64                                                                                                                      2/2

Upgraded:
  teleport-ent-updater-13.2.1-1.x86_64

Complete!
$ sudo systemctl list-timers
NEXT                        LEFT         LAST                        PASSED       UNIT                         ACTIVATES
Thu 2023-07-20 16:50:59 UTC 11min left   -                           -            teleport-upgrade.timer       teleport-upgrade.service
Thu 2023-07-20 17:40:06 UTC 1h 0min left Thu 2023-07-20 16:36:25 UTC 2min 59s ago dnf-makecache.timer          dnf-makecache.service
Fri 2023-07-21 00:00:00 UTC 7h left      Thu 2023-07-20 00:00:37 UTC 16h ago      logrotate.timer              logrotate.service
Fri 2023-07-21 00:00:37 UTC 7h left      Thu 2023-07-20 00:00:37 UTC 16h ago      systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service

4 timers listed.
Pass --all to see loaded but inactive timers, too.
```

**Comment by webvictim** [comment]
Not so fast. Installing this from scratch on CentOS Stream 8 via Teleport 13.2.3 still fails with the same symptoms:

```
[root@ip-172-31-8-8 ~]# sudo yum -y install teleport-ent-updater
Last metadata expiration check: 0:01:50 ago on Mon Jul 24 18:51:40 2023.
Dependencies resolved.
===========================================================================================================================================================================================================================================
 Package                                                           Architecture                                        Version                                                 Repository                                             Size
===========================================================================================================================================================================================================================================
Installing:
 teleport-ent-updater                                              x86_64                                              13.2.3-1                                                teleport                                               13 k

Transaction Summary
===========================================================================================================================================================================================================================================
Install  1 Package

Total download size: 13 k
Installed size: 15 k
Downloading Packages:
teleport-ent-updater-13.2.3.amd64.rpm                                                                                                                                                                       50 kB/s |  13 kB     00:00    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                                                       49 kB/s |  13 kB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                                   1/1 
  Installing       : teleport-ent-updater-13.2.3-1.x86_64                                                                                                                                                                              1/1 
  Running scriptlet: teleport-ent-updater-13.2.3-1.x86_64                                                                                                                                                                              1/1 
warning: %post(teleport-ent-updater-13.2.3-1.x86_64) scriptlet failed, signal 1

Error in POSTIN scriptlet in rpm package teleport-ent-updater
Teleport > Warning: node error. This might cause some functionalities not to work correctly.
disconnected
wait: remote command exited without exit status or exit signal
```

The issue seems to be that the updater is telling the Teleport process to restart, which causes the RPM installation to fail halfway through:
```
Jul 24 18:53:02 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:02Z INFO [AUDIT]     session.start addr.remote:156.34.165.195:52353 cluster_name:purple code:T2000I ei:0 event:session.start initial_command:[] login:root namespace:default proto:ssh server_hostname:ip-172-31-8-8.us-east-2.compute.internal server_id:050322ac-a161-4cbe-b4cb-7c97b85dbc46 teleport.internal/resource-id:7efff833-cb5c-454d-b36c-f6bf2f7baa0a session_recording:node sid:d661e57b-0637-4d19-842d-506929a532f2 size:80:25 time:2023-07-24T18:53:02.447Z uid:8580775b-dae3-4749-ada7-ba4b302507aa user:webvictim events/emitter.go:265
Jul 24 18:53:02 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:02Z INFO [SESSION:N] New party ServerContext(156.34.165.195:52353->172.31.8.8:50966, user=root, id=1) party(id=3d788b41-1f97-419a-aa0d-c150fc94f36c) joined the session with participant mode: peer. session_id:d661e57b-0637-4d19-842d-506929a532f2 srv/sess.go:1786
Jul 24 18:53:29 ip-172-31-8-8.us-east-2.compute.internal sudo[13690]:     root : TTY=pts/1 ; PWD=/root ; USER=root ; COMMAND=/bin/yum -y install teleport-ent-updater
Jul 24 18:53:30 ip-172-31-8-8.us-east-2.compute.internal sudo[13690]: pam_unix(sudo:session): session opened for user root by root(uid=0)
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal systemd[1]: Stopping Teleport Service...
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [PROC:1]    Got signal "terminated", exiting immediately. pid:10961.1 service/signals.go:86
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z ERRO [SESSION:N] failed to emit *events.SessionPrint event error:[
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: ERROR REPORT:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Original Error: *trace.ConnectionProblemError context canceled or timed out
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Stack Trace:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/events/auditwriter.go:311 github.com/gravitational/teleport/lib/events.(*AuditWriter).EmitAuditEvent
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/events/auditwriter.go:238 github.com/gravitational/teleport/lib/events.(*AuditWriter).Write
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/termmanager.go:98 github.com/gravitational/teleport/lib/srv.(*TermManager).writeToClients
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/termmanager.go:127 github.com/gravitational/teleport/lib/srv.(*TermManager).Write
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         io/io.go:429 io.copyBuffer
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         io/io.go:386 io.Copy
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/sess.go:1096 github.com/gravitational/teleport/lib/srv.(*session).launch.func2
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         runtime/asm_amd64.s:1598 runtime.goexit
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: User Message: context canceled or timed out] events/auditwriter.go:239
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z WARN             Failed to write to remote terminal: context canceled or timed out srv/termmanager.go:101
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z WARN [SESSION:N] Failed to write to session recorder, disabling session recording. error:[
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: ERROR REPORT:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Original Error: *trace.ConnectionProblemError context canceled or timed out
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Stack Trace:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/events/auditwriter.go:311 github.com/gravitational/teleport/lib/events.(*AuditWriter).EmitAuditEvent
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/events/auditwriter.go:238 github.com/gravitational/teleport/lib/events.(*AuditWriter).Write
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/termmanager.go:98 github.com/gravitational/teleport/lib/srv.(*TermManager).writeToClients
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/termmanager.go:127 github.com/gravitational/teleport/lib/srv.(*TermManager).Write
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         io/io.go:429 io.copyBuffer
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         io/io.go:386 io.Copy
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/srv/sess.go:1096 github.com/gravitational/teleport/lib/srv.(*session).launch.func2
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         runtime/asm_amd64.s:1598 runtime.goexit
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: User Message: context canceled or timed out] session_id:d661e57b-0637-4d19-842d-506929a532f2 srv/sess.go:2051
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z WARN [PROXY:AGE] Agent pool exited. error:[
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: ERROR REPORT:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Original Error: *errors.errorString context canceled
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: Stack Trace:
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/reversetunnel/agentpool.go:265 github.com/gravitational/teleport/lib/reversetunnel.(*AgentPool).run
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         github.com/gravitational/teleport/lib/reversetunnel/agentpool.go:252 github.com/gravitational/teleport/lib/reversetunnel.(*AgentPool).Start.func1
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]:         runtime/asm_amd64.s:1598 runtime.goexit
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: User Message: context canceled] localCluster: targetCluster:purple reversetunnel/agentpool.go:253
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [SESSION:N] Stopping session session_id:d661e57b-0637-4d19-842d-506929a532f2 srv/sess.go:773
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [UPLOAD:1]  File uploader is shutting down. pid:10961.1 service/service.go:2745
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [UPLOAD:1]  File uploader has shut down. pid:10961.1 service/service.go:2747
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [UPLOAD:1]  File upload completer is shutting down. pid:10961.1 service/service.go:2776
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[10961]: 2023-07-24T18:53:33Z INFO [UPLOAD:1]  File upload completer has shut down. pid:10961.1 service/service.go:2778
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal systemd[1]: teleport.service: Succeeded.
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal systemd[1]: Stopped Teleport Service.
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal systemd[1]: Started Teleport Service.
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [PROC:1]    Configured upgrade window exporter for external upgrader. kind=unit pid:13754.1 service/service.go:1010
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z WARN [CLOUDLABE] Could not fetch EC2 instance's tags, please ensure 'allow instance tags in metadata' is enabled on the instance. labels/cloud.go:140
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [PROC:1]    Connecting to the cluster purple with TLS client certificate. pid:13754.1 service/connect.go:262
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal sudo[13690]: pam_unix(sudo:session): session closed for user root
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [PROC:1]    Node: features loaded from auth server: Kubernetes:true App:true DB:true OIDC:true SAML:true AccessControls:true AdvancedAccessWorkflows:true HSM:true Desktop:true Assist:true DeviceTrust:<enabled:true >  pid:13754.1 service/connect.go:96
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [PROC:1]    Instance: features loaded from auth server: Kubernetes:true App:true DB:true OIDC:true SAML:true AccessControls:true AdvancedAccessWorkflows:true HSM:true Desktop:true Assist:true DeviceTrust:<enabled:true >  pid:13754.1 service/connect.go:96
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [INSTANCE:] Successfully registered instance client. pid:13754.1 service/service.go:2326
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [NODE:1:CA] Cache "node" first init succeeded. cache/cache.go:875
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO             debug -> starting control-stream based heartbeat. regular/sshserver.go:873
Jul 24 18:53:33 ip-172-31-8-8.us-east-2.compute.internal teleport[13754]: 2023-07-24T18:53:33Z INFO [NODE:1]    Service is starting in tunnel mode. pid:13754.1 service/service.
```

As such, the timer is not enabled:
```
[centos@ip-172-31-8-8 ~]$ sudo systemctl list-timers --all
NEXT                         LEFT          LAST                         PASSED       UNIT                         ACTIVATES
Mon 2023-07-24 18:57:08 UTC  1min 19s left n/a                          n/a          systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Mon 2023-07-24 20:12:50 UTC  1h 17min left Mon 2023-07-24 18:53:33 UTC  2min 16s ago dnf-makecache.timer          dnf-makecache.service
Tue 2023-07-25 00:00:00 UTC  5h 4min left  n/a                          n/a          unbound-anchor.timer         unbound-anchor.service

3 timers listed.
```

**Comment by bernardjkim** [comment]
Hey @fspmarshall 

Do you have any context on if the teleport service needs to be restarted on installation of the `teleport-ent-updater`? If the teleport service does not need to be restarted on installation, do you have any ideas on a work around to at least complete the installation and enable the timer?

We can see in the installation script that we restart the teleport service on installation:
```
# rpm -q --scripts teleport-ent-updater
postinstall scriptlet (using /bin/sh):
upgrade() {
    :
#!/bin/bash

# this script is run each time the teleport upgrader package is itself upgraded

set -eu

# reload systemd configuration and restart timer unit
systemctl daemon-reload
systemctl try-restart teleport-upgrade.timer

}
_install() {
    :
#!/bin/bash

# this script is run once on initial install of the teleport upgrader package

set -eu

# reload systemd configuration and start/restart units
systemctl daemon-reload
systemctl try-restart teleport.service
systemctl enable teleport-upgrade.timer
systemctl start teleport-upgrade.timer

}
if [ "${1}" -eq 1 ]
then
    # "after install" goes here
    _install
elif [ "${1}" -gt 1 ]
then
    # "after upgrade" goes here
    upgrade
fi
```

**Comment by fspmarshall** [comment]
@bernardjkim The teleport service needs to be restarted in order to detect the new env variable that is set by `etc/systemd/system/teleport.service.d/teleport-upgrader-env.conf` (`TELEPORT_EXT_UPGRADER=unit`).  This env var being set triggers the teleport agent to start exporting the maintenance window schedule.

We could just not restart it, and everything would be fine initially, but the next time the upgrader runs, if it sees that there is a new version of teleport available it will restart teleport then as part of upgrading it because it won't know the schedule, so basically your first upgrade would happen at some random time instead of during the expected window.

I think the best-case for solving this issue would be to find a way to signal teleport to start emitting the schedule value without restarting it.  I don't know off the top of my head what the best way to do that would be.

Alternatively, we could try to trigger a graceful restart where teleport exits once all current ssh sessions close... historically those have been iffy though.

Ultimately, the system will eventually self-correct, so the short-term fix might be to just remove the restart and accept that the first upgrade will happen outside the maintenance window.  It'd be simple enough to document the fact that you need to restart your `teleport.service` after installing the upgrader if you don't want this to happen.

**Comment by webvictim** [comment]
> Alternatively, we could try to trigger a graceful restart where teleport exits once all current ssh sessions close... historically those have been iffy though.

I thought this is what `kill
-HUP $(pidof teleport)` was supposed to do? It works pretty well for SSH in my experience, not so much with the app_service/db_service though.

**Comment by bernardjkim** [comment]
Looks like `kill -HUP $(pidof teleport)` has solved the issue as of `v13.3.1`.

**Comment by webvictim** [comment]
Thanks Bernard!
